### PR TITLE
Http client

### PR DESCRIPTION
**Pull request by iobaixas**

### CODE REVIEW

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -22,4 +23,6 @@
 
 sleep(1) # wait a sec for the server to be booted
 
+WebMock.disable_net_connect!(:allow_localhost => true)
```
Comment (monkeyci) [inline spec/spec_helper.rb]: Use the new Ruby 1.9 hash syntax.

File: spec/macros/support_jquery_selectors.rb [hunk 2]
```diff
@@ -11,9 +11,9 @@ def it_should_support_jquery_selectors
       it { expect(pincers.search('.bikes li:contains(\'Kona\')').count).to eq 1 }
       it { expect(pincers.search('input:text')[:id]).to eq 'name'}
       it { expect(pincers.search('input:checkbox')[:id]).to eq 'option'}
-      it { expect(pincers.search('input:radio')[:id]).to eq 'radio-1'}
+      it { expect(pincers.search('input:radio')[:id]).to eq 'first-radio'}
       it { expect(pincers.search('option:selected')[:id]).to eq 'option-2'}
-      it { expect(pincers.search('#tags input:checked')[:id]).to eq 'radio-2'}
+      it { expect(pincers.search('#tags input:checked')[:id]).to eq 'second-radio'}
```
Comment (monkeyci) [inline spec/macros/support_jquery_selectors.rb]: Space missing inside }.

File: spec/macros/support_jquery_selectors.rb [hunk 3]
```diff
@@ -11,9 +11,9 @@ def it_should_support_jquery_selectors
       it { expect(pincers.search('.bikes li:contains(\'Kona\')').count).to eq 1 }
       it { expect(pincers.search('input:text')[:id]).to eq 'name'}
       it { expect(pincers.search('input:checkbox')[:id]).to eq 'option'}
-      it { expect(pincers.search('input:radio')[:id]).to eq 'radio-1'}
+      it { expect(pincers.search('input:radio')[:id]).to eq 'first-radio'}
```
Comment (monkeyci) [inline spec/macros/support_jquery_selectors.rb]: Space missing inside }.

File: spec/macros/perform_raw_requests.rb [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        form[:extra] = 'foo'
+        response = form.submit
+
+        expect(response.content).to include 'category=private&tag=private&extra=foo'
+      end
+
+    end
+
+  end
+end
```
Comment (monkeyci) [inline spec/macros/perform_raw_requests.rb]: Final newline missing.

File: spec/macros/perform_raw_requests.rb [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+      it "should properly handle forms" do
+        form = pincers.search('form').replicate
+        form[:extra] = 'foo'
+        response = form.submit
+
+        expect(response.content).to include 'category=private&tag=private&extra=foo'
+      end
+
+    end
+
```
Comment (monkeyci) [inline spec/macros/perform_raw_requests.rb]: Extra empty line detected at method body end.

File: spec/macros/perform_raw_requests.rb [hunk 6]
```diff
@@ -0,0 +1,51 @@
+module Macros
+  def it_should_perform_raw_requests
+
```
Comment (monkeyci) [inline spec/macros/perform_raw_requests.rb]: Extra empty line detected at method body beginning.

File: spec/macros/perform_raw_requests.rb [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+      end
+
+      it "should properly handle forms" do
+        form = pincers.search('form').replicate
+        form[:extra] = 'foo'
+        response = form.submit
+
+        expect(response.content).to include 'category=private&tag=private&extra=foo'
+      end
+
```
Comment (monkeyci) [inline spec/macros/perform_raw_requests.rb]: Extra empty line detected at block body end.

File: spec/macros/perform_raw_requests.rb [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+        pincers.goto("http://localhost:#{SERVER_PORT}/index.html")
+
+        expect {
+          pincers.as_http_client.get("/resource.txt")
+        }.not_to raise_error
+      end
+    end
+
+    describe 'replicate' do
+
```
Comment (monkeyci) [inline spec/macros/perform_raw_requests.rb]: Extra empty line detected at block body beginning.

File: spec/macros/perform_raw_requests.rb [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+        pincers.goto("http://localhost:#{SERVER_PORT}/setcookie?name=sessionid&value=helloworld")
+        expect {
+          pincers.as_http_client.get("http://localhost:#{SERVER_PORT}/checkcookie?name=sessionid&value=helloworld")
+        }.not_to raise_error
+      end
+
+      it "should support relative urls" do
+        pincers.goto("http://localhost:#{SERVER_PORT}/index.html")
+
+        expect {
```
Comment (monkeyci) [inline spec/macros/perform_raw_requests.rb]: Avoid using {...} for multi-line blocks.

File: spec/macros/perform_raw_requests.rb [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+
+      it "should fail if element does not point to a resource" do
+        expect { pincers.search('label').download }.to raise_error Pincers::NavigationError
+      end
+    end
+
+    describe 'as_http_client' do
+      it "should send raw requests using cookies set by browser requests" do
+        pincers.goto("http://localhost:#{SERVER_PORT}/setcookie?name=sessionid&value=helloworld")
+        expect {
```
Comment (monkeyci) [inline spec/macros/perform_raw_requests.rb]: Avoid using {...} for multi-line blocks.

File: spec/macros/perform_raw_requests.rb (hunk 6 above)
Comment (agustinf) [inline spec/macros/perform_raw_requests.rb]: Ah mono! sh! sh!

On Wed, Oct 21, 2015 at 2:52 PM, Monkey Hound CI <notifications@github.com>
wrote:

> In spec/macros/perform_raw_requests.rb
> <https://github.com/platanus/pincers/pull/20#discussion_r42657990>:
>
> > @@ -0,0 +1,51 @@
> > +module Macros
> > +  def it_should_perform_raw_requests
> > +
>
> Extra empty line detected at method body beginning.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/platanus/pincers/pull/20/files#r42657990>.
>



-- 

[image: Platanus] <http://platan.us/>

Agustín Feuerhake
Co-fundador | Ingeniero de Software
agustin@platan.us

Platanus: (+562) 2570 9846 <%28%2B562%29%2028818025>
Las Catalpas 1588, Vitacura, Santiago, Chile

File: spec/macros/perform_raw_requests.rb [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+          pincers.as_http_client.get("http://localhost:#{SERVER_PORT}/checkcookie?name=sessionid&value=helloworld")
```
Comment (monkeyci) [inline spec/macros/perform_raw_requests.rb]: Line is too long. [115/100]

File: spec/macros/perform_raw_requests.rb [hunk 12]
```diff
@@ -0,0 +1,50 @@
+module Macros
+  def it_should_perform_raw_requests
```
Comment (monkeyci) [inline spec/macros/perform_raw_requests.rb]: Assignment Branch Condition size for it_should_perform_raw_requests is too high. [58.08/25]

File: spec/macros/handle_frames_in_example.rb [hunk 13]
```diff
@@ -19,6 +19,21 @@ def it_should_properly_handle_frames_in_example
...
+          expect(pincers.text).to include('This is the main page')
+        end
+
+        it "should load frames with target: _top in top frame" do
+          pincers.search('#top').click
+          expect(pincers.text).to include('This page is referenced from index')
+          pincers.goto frame: :top
+          expect(pincers.text).to include('This page is referenced from index')
+        end
+
```
Comment (monkeyci) [inline spec/macros/handle_frames_in_example.rb]: Extra empty line detected at block body end.

File: spec/macros/enter_data_in_example.rb [hunk 14]
```diff
@@ -39,6 +43,30 @@ def it_should_properly_enter_data_in_example
...
+
+    end
+
+    describe 'submit' do
+
+      it "should properly submit a form" do
+        pincers.search('form').submit
+        expect(pincers.text).to eq('category=private&tag=private')
+      end
+
```
Comment (monkeyci) [inline spec/macros/enter_data_in_example.rb]: Extra empty line detected at block body end.

File: spec/macros/enter_data_in_example.rb [hunk 15]
```diff
@@ -39,6 +43,30 @@ def it_should_properly_enter_data_in_example
...
+      it "should properly submit a form if submit is clicked" do
+        submit.click
+        expect(pincers.text).to eq('category=private&tag=private&button=Send+message')
+      end
+
+    end
+
+    describe 'submit' do
+
```
Comment (monkeyci) [inline spec/macros/enter_data_in_example.rb]: Extra empty line detected at block body beginning.

File: spec/macros/perform_raw_requests.rb (hunk 6 above)
Comment (blackjid) [inline spec/macros/perform_raw_requests.rb]: porque?

**Pull request merged**